Userland/Utilities/shot.cpp

```diff
@@ -1,32 +1,103 @@
 /*
  * Copyright (c) 2021, Andreas Kling <kling@serenityos.org>
+ * Copyright (c) 2021, Aziz Berkay Yesilyurt <abyesilyurt@gmail.com>
  *
  * SPDX-License-Identifier: BSD-2-Clause
  */
 
 #include <AK/Format.h>
+#include <AK/Optional.h>
 #include <LibCore/ArgsParser.h>
 #include <LibCore/DateTime.h>
 #include <LibCore/File.h>
 #include <LibGUI/Application.h>
 #include <LibGUI/Clipboard.h>
+#include <LibGUI/Painter.h>
+#include <LibGUI/Widget.h>
+#include <LibGUI/Window.h>
 #include <LibGUI/WindowServerConnection.h>
 #include <LibGfx/PNGWriter.h>
 #include <unistd.h>
 
+class SelectableLayover final : public GUI::Widget {
+    C_OBJECT(SelectableLayover)
+public:
+    SelectableLayover(GUI::Window* window)
+        : m_window(window)
+    {
+    }
+
+    virtual ~SelectableLayover() override {};
+
+    Gfx::IntRect region() const
+    {
+        return m_region;
+    }
+
+private:
+    virtual void mousedown_event(GUI::MouseEvent& event) override
+    {
+        if (event.button() == GUI::MouseButton::Left)
+            m_anchor_point = event.position();
+    };
+
+    virtual void mousemove_event(GUI::MouseEvent& event) override
+    {
+        if (m_anchor_point.has_value()) {
+            m_region = Gfx::IntRect::from_two_points(*m_anchor_point, event.position());
+            update();
+        }
+    };
+
+    virtual void mouseup_event(GUI::MouseEvent& event) override
+    {
+        if (event.button() == GUI::MouseButton::Left)
+            m_window->close();
+    };
+
+    virtual void paint_event(GUI::PaintEvent&) override
+    {
+        if (m_region.is_empty()) {
+            GUI::Painter painter(*this);
+            painter.fill_rect(m_window->rect(), m_background_color);
```

Comment:
Is it correct to repaint the background everytime this function is called?

Reply:
Eh, why not :shrug:

Reply:
I don't know much about the internals, I thought it might be costly to call everytime.